webapp-client/src/component/ContentPane.js

```diff
@@ -0,0 +1,78 @@
+import '../stylesheets/ContentPane.css';
+import errorify from '../errorify.js';
+import Item from './Item';
+import PropTypes from 'prop-types';
+import React from 'react';
+
+export default class ContentPane extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      items: [],
+    };
+    errorify(this);
+  }
+
+  componentDidMount() {
+    this.setItems(this.props);
+  }
+
+  componentDidUpdate(prevProps) {
+    if (this.props !== prevProps) {
+      this.setItems(this.props);
+    }
+  }
+
+  async setItems(props) {
+    const { categoryId, defaultCategory, userId } = props;
+    let url;
+    let response;
+    let items;
+    try {
+      if (categoryId === defaultCategory) {
+        url = process.env.REACT_APP_API_URL + '/item/getItems/' + userId;
+        response = await fetch(url);
+        items = await response.json();
+      } else {
+        url =
+          process.env.REACT_APP_API_URL +
+          '/category/getCategoryItems/' +
+          categoryId;
```

Comment:
Please add an explanation comment. Something like "Fetches all of the user's items." for the if block and something like "Fetches all of the user's items in the selected category" for the else block.

Reply:
Done